platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/module/telemetry/MapEventFactory.java

```diff
@@ -0,0 +1,139 @@
+package com.mapbox.mapboxsdk.module.telemetry;
+
+import android.content.Context;
+import android.content.res.Configuration;
+import android.net.wifi.WifiInfo;
+import android.net.wifi.WifiManager;
+import android.support.annotation.NonNull;
+import android.telephony.TelephonyManager;
+import android.text.TextUtils;
+import android.util.DisplayMetrics;
+import android.view.WindowManager;
+
+import com.mapbox.android.telemetry.Event;
+import com.mapbox.android.telemetry.TelemetryUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class MapEventFactory {
+  private static final String LANDSCAPE = "Landscape";
+  private static final String PORTRAIT = "Portrait";
+  private static final String NO_CARRIER = "EMPTY_CARRIER";
+  private static final int NO_NETWORK = -1;
+
+  private static final Map<Integer, String> ORIENTATIONS = new HashMap<Integer, String>() {
+    {
+      put(Configuration.ORIENTATION_LANDSCAPE, LANDSCAPE);
+      put(Configuration.ORIENTATION_PORTRAIT, PORTRAIT);
+    }
+  };
+
+  private final Context appContext;
+
+  MapEventFactory(Context appContext) {
+   this.appContext = appContext;
+  }
+
+  Event buildMapLoadEvent() {
+    String userId = TelemetryUtils.retrieveVendorId();
```

Comment:
cannot be accessed from outside the package

Reply:
Currently it is only used in `TelemetryImpl`, do you want to use this method outside the package?

Reply:
You mean `TelemetryUtils.retrieveVendorId()` can't be accessed? Already updated it to public in telemetry sdk.

Reply:
Appologies, I missed the reference to https://github.com/mapbox/mapbox-events-android/pull/373